### PR TITLE
Make base_badge_price actually store correct price

### PR DESCRIPTION
**Pull request by kitsuta**
Fixes MAGDEV-254.

One of the entire points of base_badge_price was to allow us to store what the badge WOULD have cost if it were a regular attendee badge... a purpose we completely undermined by putting discounted and overridden prices in the calculation we used for base_badge_price. Most noticeably, this led us to charging dealers whose apps had been rejected only the dealer price, not the attendee price at the time they applied. We've instead moved ALL discounts and overrides to calculate_badge_cost, so they get applied to any price except the "base" price.

Thank goodness for unit tests.

### CODE REVIEW

**Comment by RobRuana** [comment]
Okay, I'll be honest. I have no idea how our pricing system works. We've got a bunch of different cost properties used in a bunch of different places for different purposes. I am scared that changing something in one place will have ramifications in other places.

For example, we set `self.base_badge_price = self.new_badge_cost` in a presave adjustment:
https://github.com/magfest/ubersystem/blob/09c70709a029ac1d761c14f4cc8fe96c8c5252fd/uber/models/attendee.py#L443

Is that what we want to do? If so, have we been saving the wrong base badge price all this time? 

Basically, I'm saying I need to sit down for an hour or so and work through how and when badge prices are calculated before I'm comfortable merging this.

**Comment by kitsuta** [comment]
Yes, that is what we want to do and yes, we've been saving the wrong badge price this whole time.

Wanting to think over it is totally fair! Our unit tests found a few bugs with my first implementation, and the unit test I wrote then found some more. It's pretty complex. 😓 I'd say that, although it will cause some issues later on, we can launch dealer reg without this change. Since badge prices won't change, we can easily(ish) fix the data later.
